### PR TITLE
Add acceptance test for copying public link URL and assert shared resource

### LINKED ISSUE #2525: Acceptance tests for copying public link URL
## Add acceptance test for copying pubic link URL 
Phoenix app provides a copy link button for public links created for any resource. Acceptance tests should be added considering its behavior.

### PR DESCRIPTION
**Pull request by kiranparajuli589**
## Description
In phoenix, when we create a public link for some resource, we get a button that copies the public link uri in clipboard and then we can paste it on browser and view shared items. WebUI acceptance test for this scenario has been added.

## Related Issue

- Fixes #2525

## How Has This Been Tested?

:robot: 
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] ...

### CODE REVIEW

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingPublic</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/6525/

<p>





</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>PrivateLinks</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/6660/

<p>







</p></details>

File: tests/acceptance/stepDefinitions/publicLinkContext.js [hunk 1]
```diff
@@ -182,3 +182,21 @@ When('the user closes the public link details sidebar', function () {
...
+    .closeSidebar(100)
+    .openPublicLinkDialog(resource)
+  return client.page.FilesPageElement.publicLinksDialog()
+    .clickCopyLinkURI(linkName)
+})
+
+When('the user navigates to the copied public link using the webUI', function () {
+  return client
+    .getClipBoardContent(function (url) {
+      setTimeout(function () {
```
Comment (dpakach) [inline tests/acceptance/stepDefinitions/publicLinkContext.js]: Please specify the reason behind using this pattern as comment.

File: tests/acceptance/features/webUISharingPublic/shareByPublicLink.feature [hunk 2]
```diff
@@ -644,3 +644,11 @@ Feature: Share by public link
     When the user uploads file "lorem.txt" keeping both new and existing files using the webUI
     Then file "lorem.txt" should be listed on the webUI
     And file "lorem (2).txt" should be listed on the webUI
+
+  Scenario: user browses to public link share using copy link button
+    Given user "user1" has shared folder "simple-folder" with link with "read" permissions
+    And user "user1" has logged in using the webUI
+    When the user copies the public link named "{}" of the folder "simple-folder" using the webUI
```
Comment (dpakach) [inline tests/acceptance/features/webUISharingPublic/shareByPublicLink.feature]:
```suggestion
    When the user copies the url of public link named "{}" of folder "simple-folder" using the webUI
```
I think this sounds better.

File: tests/acceptance/stepDefinitions/publicLinkContext.js (hunk 1 above)
Comment (kiranparajuli589) [reply tests/acceptance/stepDefinitions/publicLinkContext.js]: :rocket:

File: tests/acceptance/features/webUISharingPublic/shareByPublicLink.feature (hunk 2 above)
Comment (kiranparajuli589) [reply tests/acceptance/features/webUISharingPublic/shareByPublicLink.feature]: :rocket:

File: tests/acceptance/pageObjects/FilesPageElement/publicLinksDialog.js [hunk 3]
```diff
@@ -470,6 +485,10 @@ module.exports = {
     dateTimeCancelButton: {
       selector: '//div[@class="vdatetime-popup__actions"]/div[.="Cancel"]',
       locateStrategy: 'xpath'
+    },
+    publicLinkURLCopyButton: {
```
Comment (individual-it) [inline tests/acceptance/pageObjects/FilesPageElement/publicLinksDialog.js]: maybe place together with `publicLinkDeleteButton`

File: tests/acceptance/stepDefinitions/sharingContext.js [hunk 4]
```diff
@@ -337,7 +336,7 @@ Given(
       SHARE_TYPES.public_link,
       permissions,
       null,
-      { password: password }
+      { password }
```
Comment (individual-it) [inline tests/acceptance/stepDefinitions/sharingContext.js]: why that change?

File: tests/acceptance/stepDefinitions/sharingContext.js [hunk 5]
```diff
@@ -116,8 +116,7 @@ const createPublicLink = function (sharer, data) {
     {
       password,
       expireDate
-    }
-  )
+    })
```
Comment (individual-it) [inline tests/acceptance/stepDefinitions/sharingContext.js]: why that change?

File: tests/acceptance/stepDefinitions/publicLinkContext.js [hunk 6]
```diff
@@ -182,3 +182,23 @@ When('the user closes the public link details sidebar', function () {
...
+When('the user copies the url of public link named {string} of file/folder/resource {string} using the webUI', async function (linkName, resource) {
+  await client.page.FilesPageElement
+    .filesList()
+    .closeSidebar(100)
+    .openPublicLinkDialog(resource)
+  return client.page.FilesPageElement.publicLinksDialog()
+    .clickCopyLinkURI(linkName)
+})
+
+When('the user navigates to the copied public link using the webUI', function () {
```
Comment (individual-it) [inline tests/acceptance/stepDefinitions/publicLinkContext.js]: this is the same as `When('the user navigates to the copied private link using the webUI', function () {` please DRY
```suggestion
When('the user navigates to the link in clip-board using the webUI', function () {
```

File: tests/acceptance/pageObjects/FilesPageElement/publicLinksDialog.js [hunk 7]
```diff
@@ -384,6 +384,21 @@ module.exports = {
         message = result.value
       })
       return message
+    },
+    /**
+     * clicks the 'copy-public-link-uri' button of a public link
+     *
+     * @param {string} linkName Name of the public link whose URL is to be copied
+     */
+    clickCopyLinkURI: function (linkName) {
```
Comment (individual-it) [inline tests/acceptance/pageObjects/FilesPageElement/publicLinksDialog.js]: don't use the word `click` on a mobile device we might not "click"
```suggestion
    copyPublicLinkURI: function (linkName) {
```

File: tests/acceptance/pageObjects/FilesPageElement/publicLinksDialog.js (hunk 3 above)
Comment (kiranparajuli589) [reply tests/acceptance/pageObjects/FilesPageElement/publicLinksDialog.js]: :rocket:

File: tests/acceptance/stepDefinitions/sharingContext.js (hunk 5 above)
Comment (kiranparajuli589) [reply tests/acceptance/stepDefinitions/sharingContext.js]: :rocket:

File: tests/acceptance/stepDefinitions/sharingContext.js (hunk 4 above)
Comment (kiranparajuli589) [reply tests/acceptance/stepDefinitions/sharingContext.js]: :rocket:

File: tests/acceptance/pageObjects/FilesPageElement/publicLinksDialog.js (hunk 7 above)
Comment (kiranparajuli589) [reply tests/acceptance/pageObjects/FilesPageElement/publicLinksDialog.js]: :rocket:

File: tests/acceptance/pageObjects/FilesPageElement/publicLinksDialog.js (hunk 3 above)
Comment (kiranparajuli589) [reply tests/acceptance/pageObjects/FilesPageElement/publicLinksDialog.js]: :rocket:

File: tests/acceptance/stepDefinitions/publicLinkContext.js (hunk 6 above)
Comment (kiranparajuli589) [reply tests/acceptance/stepDefinitions/publicLinkContext.js]: :rocket:

**Pull request merged**